### PR TITLE
ProjectPage Details api

### PR DESCRIPTION
**Pull request by LeenMass**

### CODE REVIEW

**Comment by tawfiknasser** [comment]
@LeenMass  did you notice the issue with the CI ?

File: controllers/general.controller.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-const catchAsync = require('../utils/catchAsync')
-const ApiError = require('../utils/ApiError')
 
+const catchAsync = require("../utils/catchAsync")
+const ApiError = require("../utils/ApiError")
+const db = require("../database/connection");
```
Comment (tawfiknasser) [inline controllers/general.controller.js]: is db used in this file ?

File: controllers/general.controller.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+const showProjects=require("../database/models/Project")
```
Comment (tawfiknasser) [inline controllers/general.controller.js]: `showProject` is defined but not used?

**Review by tawfiknasser — CHANGES_REQUESTED** [review]

File: controllers/general.controller.js (hunk 2 above)
Comment (tawfiknasser) [reply controllers/general.controller.js]: @Karyum @MuhammadAw90 @LeenMass prettier is not used here!

File: controllers/projects.controller.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 const catchAsync = require('../utils/catchAsync')
 const ApiError = require('../utils/ApiError')
-const { getProjects } = require('../database/models/Project')
+const { getProjects,getProjectsDetails } = require('../database/models/Project')
```
Comment (tawfiknasser) [inline controllers/projects.controller.js]: @Karyum @MuhammadAw90 @LeenMass Prettier is not used here!

File: controllers/projects.controller.js [hunk 4]
```diff
@@ -16,6 +19,23 @@ const getProjectsController = catchAsync(async (req, res) => {
   }
 })
 
+const showProjectsData=catchAsync(async(req,res)=>{
```
Comment (tawfiknasser) [inline controllers/projects.controller.js]: @Karyum @MuhammadAw90 @LeenMass  Prettier is not used here !

File: database/models/Project.js [hunk 5]
```diff
@@ -25,6 +24,13 @@ const editProject = async (
 
 const getProjects = async () => {
   const projectTable = await db.query(`SELECT * FROM project`)
-  return projectTable.rows
+
+  return Promise.resolve(projectTable.rows) 
+
+}
+
+const getProjectsDetails=async()=>{
```
Comment (tawfiknasser) [inline database/models/Project.js]: @Karyum @LeenMass @MuhammadAw90 Prettier is defiantly not used here!

File: database/models/Project.js [hunk 6]
```diff
@@ -25,6 +24,13 @@ const editProject = async (
...
+  const data = await db.query("SELECT project.project_name,(SELECT COUNT(*) FROM iteration where project_id=project.id) as iteration_number, (SELECT COUNT(*) FROM scrum where project_id=project.id) as scrum_number,(SELECT COUNT(*) FROM employee WHERE employee.project_id = project.id) as employee_number FROM project");
+  return data.rows
 }
-module.exports = { createProject, editProject, getProjects }
+module.exports = { createProject, editProject, getProjects,getProjectsDetails }
```
Comment (tawfiknasser) [inline database/models/Project.js]: and here. no prettier is used
@LeenMass @MuhammadAw90 @Karyum

**Comment by Karyum** [comment]
@tawfiknasser I know we are talking about prettier in a different PR, but why am I mentioned in the PR of a different project of a different pull request ?
I can fix the errors if that's what you meant

**Comment by LeenMass** [comment]
all the comments resolved 
@tawfiknasser please re-review

**Review by tawfiknasser — APPROVED** [review]

**Comment by tawfiknasser** [comment]
@LeenMass Looks great! 
can you have more two approves from your team mates? we need others to be more engaged.

**Review by MuhammadAw90 — APPROVED** [review]
Good job!

**Review by Karyum — APPROVED** [review]

**Pull request merged**